src/redux/books/books.js

```diff
@@ -1,11 +1,31 @@
-import BookInfo from './BookInfo';
-import BookInput from './BookInput';
+const ADD_BOOK = 'bookstore/books/ADD_BOOK';
+const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
 
-const Books = () => (
-  <div className="booksContainer">
-    <BookInfo />
-    <BookInput />
-  </div>
-);
+const initialState = [];
 
-export default Books;
+export const bookAdded = (payload) => ({
+  type: ADD_BOOK,
+  payload,
+});
+
+export const bookRemoved = (payload) => ({
+  type: REMOVE_BOOK,
+  payload,
+});
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return [
+        ...state,
+        action.payload,
+      ];
+
+    case REMOVE_BOOK:
+      return state.filter((book) => book.id !== action.payload);
+    default:
+      return state;
+  }
+};
```

Comment:
- [ ] In your reducer switch statement, kindly add a default case in which if the action is different from the defined ones, it returns the state with no changes, as explained on the requirements:

![image](https://user-images.githubusercontent.com/57421823/152290010-a065ac64-7f5e-4575-86a8-b2d485b674ec.png)

In other words, just kindly please add a default case in which the unchanged state is returned.

That way, your reducer will be a lot more consistent, rock-solid, and bug-proof! 💪

Reply:
The default is already there. Should I add another default apart from the default already there?

Reply:
You are VERY right, of course, the default case is actually there.  Thanks for pointing this out.

Also, you even added an empty line before the default case in order to make it clearer in your last commit, (consistent indentation and spacing help us a lot) and I thank you for that. 🙏 

Approving now, since your project works perfectly.

Reply:
You are VERY right, of course, the default case is actually there. Thanks for pointing this out.

Also, you even added an empty line before the default case in order to make it clearer in your last commit, (consistent indentation and spacing help us a lot) and I thank you for that. 🙏

Approving now, since your project works perfectly.